route/build/src/write_restart_pio.f90

```diff
@@ -959,37 +982,37 @@ SUBROUTINE write_state_nc(fname,                &   ! Input: state netcdf name
  call write_history_state(ierr, cmessage)
  if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
 
+ ! close netCDF
+ call closeFile(pioFileDescState, restartOpen)
+
  ! clean decomposition data
```

Comment:
My bet is that the pio File descriptor option, actually then points to the pioSystem that's associated with the pioFileDesc. One problem is if the pioFileDesc is freed before you do that. I also had problems with the Nag compiler if you didn't close pioFileDesc, before doing the decomp free.  Which just tells me in this case we should always use pioSystem. We could bring this up with Jim, but since we have it working I'm not sure how important it is. But, certainly not bad to do...